### PR TITLE
Use ESlint in replace of jshint

### PR DESCRIPTION
**Pull request by davidcheung**
- Linting generated code
- Use ESlint in replace of jshint

### CODE REVIEW

File: lib/services.js [hunk 1]
```diff
@@ -2,7 +2,12 @@ var fs = require('fs');
 var ejs = require('ejs');
 
 ejs.filters.q = function(obj) {
-  return JSON.stringify(obj, null, 2 );
+  // to handle strings being returned with double-quotes
+  // but linter requires single quotes for strings
+  if (typeof obj == 'string' && !/\\'/.test(obj))
+    return '\'' + obj + '\'';
+  else
+    return JSON.stringify(obj, null, 2);
```
Comment (davidcheung) [inline lib/services.js]: not sure if this is the best approach
I have modified the `ejs` filter so string gets wrapped in single quote instead of double quote that JSON.stringify does

Comment (superkhau) [inline lib/services.js]: do not change the code logic, the only difference should be the space at the end of the 2

```
return JSON.stringify(obj, null, 2);
```

The other stuff you're changing is untested and should be done in a separate PR.

File: lib/services.template.ejs [hunk 2]
```diff
@@ -9,17 +10,18 @@ if (typeof module !== 'undefined' && typeof exports !== 'undefined' && module.ex
   module.exports = <%-: moduleName | q %>;
 }
```
Comment (superkhau) [inline lib/services.template.ejs]: all stuff inside should be indented one more level

```
if (typeof module !== 'undefined' && typeof exports !== 'undefined' &&
  module.exports === exports) {
     // Export the *name* of this Angular module
     // Sample usage:
     //
     //   import lbServices from './lb-services';
     //   angular.module('app', [lbServices]);
     //
     module.exports = <%-: moduleName | q %>;
   }
```

Comment (superkhau) [inline lib/services.template.ejs]: However, the indenting is not so important to fix now (we can do those when we actually modify that line of code.

File: package.json [hunk 3]
```diff
@@ -5,8 +5,8 @@
   "main": "index.js",
   "scripts": {
     "prepublish": "node ./apidocs/describe-builtin-models.js",
-    "pretest": "jshint .",
-    "test": "node ./test.e2e/test-server.js node node_modules/karma/bin/karma start --single-run --browsers PhantomJS"
+    "test": "node ./test.e2e/test-server.js node node_modules/karma/bin/karma start --single-run  --browsers PhantomJS",
+    "posttest": "npm run prepublish && eslint ."
```
Comment (superkhau) [inline package.json]: why do you run prepublish here? the original only ran `eslint .`?

File: lib/services.template.ejs (hunk 2 above)
Comment (superkhau) [inline lib/services.template.ejs]: Don't bother with the indenting for now.

**Comment by superkhau** [comment]
Please remove the new code and put those in separate PRs, otherwise LGTM.

File: package.json (hunk 3 above)
Comment (davidcheung) [inline package.json]: prepublish creates a local file `loopback-core.js` (aka `lb-services.js`), goal of this is to lint the generated code, because we don't lint the `ejs` file itself

File: lib/services.js (hunk 1 above)
Comment (davidcheung) [inline lib/services.js]: In that case we will need the other PR to land before this, it will cause hundreds of Lint errors on the generated code. 

@superkhau right now I have two somewhat closely related tasks in this PR
- linting generated code
- changing the linter to eslint
so do you think I should separate them into PR as such? 
please advise

Comment (superkhau) [inline lib/services.js]: It's fine, just update the comment with a clearer description. You have the why, but also include the what (as in what is the context - this code case is here for the generated file, blah blah).

File: package.json (hunk 3 above)
Comment (superkhau) [inline package.json]: Thats fine then, too bad we can't add comments to JSON. ;) Maybe add a comment in the readme regarding posttest and why we run prepublish.

Comment (superkhau) [inline package.json]: Another possiblity is to add a "generate-local-file" script and have posttest and prepublish call `npm run generate-local-file`.

**Comment by davidcheung** [comment]
Going to have linting generated code in a separate PR instead.

**Comment by superkhau** [comment]
LGTM

**Comment by superkhau** [comment]
@davidcheung Delete the merged branch?

**Comment by bajtos** [comment]
@davidcheung @superkhau I am rather worried about the indentation changes you made in the `ejs` template. The old version was finely tuned to ensure the generated code looks as nice as possible. Are you sure the new version looks as good as the old one?

**Comment by bajtos** [comment]
Ignore my last comment, I `diffed` the old and the new version and it looks okay. Good job!
